libraries/lib-project/CMakeLists.txt

```diff
@@ -0,0 +1,26 @@
+#[[
+A library defining the top Project object, which is little more than a
+container for attached objects built by registered factories, and an event
+handler.
+
+Also a global array of existing projects.
+
+Also ProjectStatus, an abstraction of the status bar, just holding some strings
+and formatter functions, and emitting events when changed.
+]]#
+
+set( SOURCES
+   Project.cpp
+   Project.h
+   ProjectStatus.cpp
+   ProjectStatus.h
+)
+set( LIBRARIES
+   lib-registries-interface
```

Comment:
I think all dependencies should be private here

Reply:
Why?

Transitive dependencies simplify many other CMakeLists.txt files and the generated graph of dependencies.

I make the exception only for wxBase for certain other reasons.

My intention (although I haven’t yet reimplemented it with all the other build system changes) was that some libraries would use only the wxBase subset, and even be forced to compile with only a subset of that — by predefining on the command line the inclusion guard symbols of the headers I wanted to disallow, such as wx/app.h (in wxBase but using the event loop, so not allowed for platform neutrality)

In that case only, making the wxBase library a private dependency was needed to avoid breaking compilation of higher level things that are allowed to use those headers.

Reply:
That is: wxBase was meant to be a CMake target enforcing this restricted view of the wxBase library to the really platform neutral part (no event loop) so I can be sure libraries using it are really neutral.

Reply:
For the sake of the correct dependency graph, I really think that every dependency should be listed explicitly, not implicitly by some other dependency. The linker will not drop such dependencies. The compiler will have to deal with the larger set of include directories. So in the end the dependency graph won't reflect the actual state of the dependencies.

To clarify my case: consider a library, that depends on `lib-project`. It should explicitly list the dependency on `lib-registries-interface` if it uses the library API internally, rather than hide this dependency somewhere in the middle of the graph. This way it will be clear, which library actually depends on `lib-registries-interface`.

Reply:
I simply disagree.  I want my generated graphs to have paths of dependency of A on B, which is good enough, but not a clutter of too many crossing edges making them too difficult to understand.

Please do not insist on this before approving the commit.